src/service.js

```diff
@@ -189,6 +190,27 @@ module.exports = function(mixinOptions) {
 							return Array.isArray(dataLoaderKey)
 								? await dataLoader.loadMany(dataLoaderKey)
 								: await dataLoader.load(dataLoaderKey);
+						} else if (fileUploadArg != null && args[fileUploadArg] != null) {
+							if (Array.isArray(args[fileUploadArg])) {
+								return await Promise.all(
+									args[fileUploadArg].map(async uploadPromise => {
+										const {
+											createReadStream,
+											...$fileInfo
+										} = await uploadPromise;
+										const stream = createReadStream();
+										return await context.ctx.call(actionName, stream, {
+											meta: { $fileInfo },
```

Comment:
@icebob I'll ask that you weigh in here.  It seems to me that the `$` convention is used in many places within moleculer to signify something internal about moleculer.  Since this property is being added to the `meta` automatically, it seems like a good convention to follow here.  The only way I can see to get the streaming data in AND the file information would be to make use of `meta`.

Reply:
Yes, the `ctx.meta.$fileInfo` is good to follow this convention.